proposals/cli_schema.md

```diff
@@ -0,0 +1,246 @@
+      - name: rulename01
+        # values themselves do not need to be validated other than for YAML syntax
+        values:
+          # values must support all types (string, int, float, bool, object, etc.)
+          imageSize: "3Gi"
+          # values may be in JMESPath expression format (dot-separated fields)
+          namespacefilters.data.exclude: "[\"default\", \"test\"]"
+          # a given value must also support an object like below
+          imageData:
+            labels:
+              org.opencontainers.image.source: "https://github.com/kyverno/kyverno-examples"
+              org.opencontainers.image.foo: "bar"
+        # if a policy is written with a namespaceSelector, it falls under the rule.
+        # See https://github.com/kyverno/kyverno/tree/main/test/cli/test/any-namespaceSelector
+        namespaceSelector:
+        - name: test1
+          labels:
+            foo.com/managed-state: managed
+      # values which pertain to resources only, commonly request.operation
+      # can be specified under this resources object. The name must match some
+      # name that is found in spec.resources[].
+      resources:
+      - name: nginx-demo2
+        values:
+          request.operation: UPDATE
+        # the userInfo field holds info on who/what is responsible for the resource. It is grouped
+        # here, just like some values, because they accompany the resource itself and not
+        # the rule. See output of `kubectl explain policy.spec.rules.match.subjects` for details
+        userInfo:
+          clusterRoles:
+          - cluster-admin
+      - name: nginx-demo3
+        userInfo:
+          roles:
+          - nsoperator
+      - name: nginx-demo4
+        userInfo:
+          subjects:
+          - name: chip
+            kind: User
+          - name: devops
+            kind: Group
+      - name: nginx-demo5
+        userInfo:
+          subjects:
+          - name: kyverno
+            kind: ServiceAccount
+            # Namespace is only applicable to ServiceAccount and should produce
+            # a validation error if present on User or Group.
+            namespace: kyverno
+    - name: very-attestations01
+      rules:
+      - name: verifyimagesrulename
+        # there will be an attestations object under rules[] which allows specifying variables
+        # related to verifyImage rule types that need attestations. The below is a work in progress and
+        # will need community input.
+        attestations:
+        - predicateType: https://example.com/CycloneDX/v1
+          predicateResource: my-sbom.json
+        - predicateType: https://example.com/CodeReview/v1
+          predicateResource: my-codereview.json
+        values:
+          foo: bar
+```
+
+## Implementation
+
+The implementation will be done by an LFX mentee supervised by one or more Kyverno maintainers with assistance provided, as necessary, by the broader contributor community.
+
```

Comment:
Can we please further elaborate on the intended design?

Are we planning on creating an OpenAPIV3 schema? How will it be used?

Reply:
I am not competent to elaborate on the implementation section. This will either need @vyankyGH's input or input from other developer-maintainers.

Reply:
@samj1912 do you have any suggestions here?

Reply:
Bump @samj1912

Reply:
It's possible to validate without a cluster. Multiple different golang libs exist for the purpose, e.g.: https://github.com/getkin/kin-openapi